chess_model.rb

```diff
@@ -290,5 +217,44 @@ def pawn_move(piece)
     valid_moves
   end
 
+  def coordinate_to_object(coordinate)
+    @board[coordinate[0]][coordinate[1]]
+  end
+
+  def valid_spot?(coord, player_color)
+    piece = @board[coord[0]][coord[1]]
+    piece != nil && piece.color == player_color && coord[0].between?(0, 7) && coord[1].between?(0, 7) ? true : false
+  end
+
+  def to_s
+    counter = 8
+    piece_picture = ""
+    @board.reverse.each do |row|
+      row.each_with_index do |piece, piece_index|
+        piece_picture += "#{counter} " if piece_index == 0
+        piece_picture += "\u265C\s" if piece.is_a?(Rook) && piece.color == "black"
```

Comment:
Perhaps all this logic about which unicode character to display could go into the Piece classes?